models/keyword.go

```diff
@@ -22,14 +54,29 @@ func GetKeywordsBy(condition map[string]interface{}) ([]Keyword, error) {
 	return keywords, nil
 }
 
-func SaveKeywords(keywords []Keyword) ([]Keyword, error) {
-	// Insert bulk data
-	result := db.GetDB().Create(&keywords)
+func SaveKeyword(keyword Keyword, tx ...*gorm.DB) (Keyword, error) {
+	var cnx *gorm.DB
+	if tx != nil {
+		cnx = tx[0]
+	} else {
+		cnx = db.GetDB()
+	}
+
+	result := cnx.Create(&keyword)
 	if result.Error != nil {
-		return nil, result.Error
+		return Keyword{}, result.Error
```

Comment:
Why not returning nil as before?

Reply:
Because struct type not allow to return nil
```go
cannot convert nil (untyped nil value) to Keyword
```

Reply:
Oh yes, indeed 😓